Makefile

```
@@ -19,13 +22,19 @@ help:
 	@echo '    build           Build package'
 	@echo '    clean           Clean directories'
 
-.PHONY: setup
-setup:
+venv:
+	test -d $@ || mkdir -m 755 ./$@
```

Why do we need to create the directory with explicit permissions?

---

Rationale:
Enumerated-Falsehood-authorization (EFA) 24 and EFA 32 [see enumerated list](https://www.osohq.com/post/falsehoods-about-authorization)

The alternative is we must check the directory for permission errors, (for example some shells set default `umask 137` (readonly + no-search/no-execute) for security best-practice)

If you'd rather the check first approach, then consider:
```suggestion
	test -d $@ || mkdir -v ./$@
	test -x $@ || chmod -v 755 $@
```
(but then we should probably consider symbolic links too, and perhaps use `mktemp`, etc. :shrug:)

---

While I partially understand that there might be cases where the permissions do not match, but I would expect users to be aware of this. I personally have never seen such a case.

Which widely used shell requires modifying the permissions afterwards?

---

:speak_no_evil: I apologise for the confusion. My wording was more misleading than I meant; Let me clarify: "some shells" should be "some shell environments"

I would also expect _most_ users to be aware of their environment's permissions. However, I also don't think we should leave such an easy automation to first-time users of the project.

> Automation is my zen

Hopefully this helps clarify. Is there anything about this code you still have any questions or concerns about?

---

While I am still not fully convinced (and honestly never heard of Oso and how trustworthy their list is), I am okay with including these changes.